api/src/main/java/org/openmrs/CohortMembership.java

```diff
@@ -112,7 +112,8 @@ public Date getEndDate() {
 	 * @param endDate
 	 */
 	public void setEndDate(Date endDate) {
-		this.endDate = endDate;
+		this.endDate = endDate != null ? (Date) endDate.clone() : null; 
+	}
 	}
```

Comment:
Hii @ribhavsharma, this extra brace over here is causing the build to fail, kindly correct it :)